RecoParticleFlow/PFClusterProducerAlpaka/plugins/LegacyPFClusterProducer.cc

```diff
@@ -0,0 +1,158 @@
+        const std::string& algoac = acConf.getParameter<std::string>("algoName");
+        _positionCalc = PFCPositionCalculatorFactory::get()->create(algoac, acConf, cc);
+      }
+
+      if (pfcConf.exists("allCellsPositionCalc")) {
+        const edm::ParameterSet& acConf = pfcConf.getParameterSet("allCellsPositionCalc");
+        const std::string& algoac = acConf.getParameter<std::string>("algoName");
+        _allCellsPositionCalc = PFCPositionCalculatorFactory::get()->create(algoac, acConf, cc);
+      }
+    }
+  }
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<edm::InputTag>("src");
+    desc.add<edm::InputTag>("PFRecHitsLabelIn");
+    desc.add<edm::ESInputTag>("pfClusterParams");
+    desc.add<edm::InputTag>("recHitsSource");
+    desc.setAllowAnything();
+    descriptions.addWithDefaultLabel(desc);
+  }
+
+  // the actual algorithm
+  std::unique_ptr<PFCPositionCalculatorBase> _positionCalc;
+  std::unique_ptr<PFCPositionCalculatorBase> _allCellsPositionCalc;
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  const edm::EDGetTokenT<reco::PFClusterHostCollection> alpakaPfClustersToken;
+  const edm::EDGetTokenT<reco::PFRecHitFractionHostCollection> alpakaPfRHFractionToken;
+  const edm::EDGetTokenT<reco::PFRecHitHostCollection> InputPFRecHitSoA_Token_;
+  const edm::ESGetToken<reco::PFClusterParamsAlpakaESDataHost, PFClusterParamsAlpakaESRecord> pfClusParamsToken;
+  const edm::EDPutTokenT<reco::PFClusterCollection> legacyPfClustersToken;
+  const edm::EDGetTokenT<reco::PFRecHitCollection> _recHitsLabel;
+
+  int nRH = 0;
+};
+
+void LegacyPFClusterProducer::produce(edm::Event& event, const edm::EventSetup& setup) {
+  const reco::PFRecHitHostCollection& pfRecHits = event.get(InputPFRecHitSoA_Token_);
+
+  reco::PFClusterHostCollection const& clusterSoA = event.get(alpakaPfClustersToken);
+  reco::PFRecHitFractionHostCollection const& fractionSoA = event.get(alpakaPfRHFractionToken);
+
+  auto const& alpakaPfClusters = clusterSoA.const_view();
+  auto const& alpakaPfRhFrac = fractionSoA.const_view();
+
+  nRH = pfRecHits.view().size();
+  reco::PFClusterCollection out;
+  out.reserve(nRH);
+
+  auto const rechitsHandle = event.getHandle(_recHitsLabel);
+
+  std::vector<int> seedlist1;
+  std::vector<int> seedlist2;
+  seedlist1.reserve(alpakaPfClusters.nSeeds());
+  for (int i = 0; i < alpakaPfClusters.nSeeds(); i++) {
+    seedlist1.push_back(alpakaPfClusters[i].pfc_seedRHIdx());
+  }
+
+  // Build PFClusters in legacy format
+  std::unordered_map<int, int> nTopoSeeds;
+
+  for (int i = 0; i < alpakaPfClusters.nSeeds(); i++) {
+    nTopoSeeds[alpakaPfClusters[i].pfc_topoId()]++;
+  }
+
+  for (int i = 0; i < alpakaPfClusters.nSeeds(); i++) {
+    const reco::PFRecHitRef& refhitTest = reco::PFRecHitRef(rechitsHandle, alpakaPfClusters[i].pfc_seedRHIdx());
+    if (refhitTest->energy() == 0)
+      printf("PROBLEM\n");
```

Comment:
Either use `LogError`/`LogWarning` or throw an exception.

Reply:
This was a loop from development, it is now removed.